File: frontend-ts/db/conversations.ts
import { SupabaseClient } from "@supabase/supabase-js";

export const dbInsertConversation = async (
    supabase: SupabaseClient,
    data: IConversation,
) => {
    const { error } = await supabase.from("conversations").insert([data]);
    if (error) {
        throw error;
    }
};

// Function to get conversations of the last 7 days
export const dbGetConversation = async (
    supabase: SupabaseClient,
    userId: string,
) => {
    // Get the date 7 days ago from today
    const sevenDaysAgo = new Date();
    sevenDaysAgo.setDate(sevenDaysAgo.getDate() - 7);

    // Convert the date to ISO string for comparison
    const sevenDaysAgoISO = sevenDaysAgo.toISOString();

    // csort by created_at
    const { data, error } = await supabase
        .from("conversations")
        .select("*")
        .eq("user_id", userId)
        .eq("role", "user")
        .gte("created_at", sevenDaysAgoISO);
    // .order("created_at", { ascending: true });

    if (error) {
        throw error;
    }
    return data;
};

// Function to get conversations of the last 10 messages
export const dbGetRecentMessages = async (
    supabase: SupabaseClient,
    userId: string,
    toyId: string,
) => {
    const { data, error } = await supabase
        .from("conversations")
        .select("*")
        .eq("user_id", userId)
        .eq("toy_id", toyId)
        .order("created_at", { ascending: false })
        .limit(20);

    if (error) {
        throw error;
    }
    return data;
};

File: backend-py/app/db/conversations.py
from email import message

from app.db.supabase import create_supabase_client


async def get_msgs(user_id: str, toy_id: str):
    supabase = create_supabase_client()
    # get last 30 messages from database from last 10 messages
    try:
        messages = (
            supabase.table("conversations")
            .select("*")
            .eq("user_id", user_id)
            .eq("toy_id", toy_id)
            .order("created_at", desc=True)
            .limit(30)
            .execute()
        )
        return messages
    except Exception as e:
        return {"error": str(e)}


def add_msg(
    toy_id: str,
    user_id: str,
    role: str,
    content: str,
    metadata: dict,
    chat_group_id: str,
    is_sensitive: bool = False,
):
    supabase = create_supabase_client()
    # add message to database
    try:
        supabase.table("conversations").insert(
            {
                "toy_id": toy_id,
                "user_id": user_id,
                "role": role,
                "content": content,
                "metadata": metadata,
                "chat_group_id": chat_group_id,  # session id
                "is_sensitive": is_sensitive,
            }
        ).execute()
        return {"message": "Message added"}
    except Exception as e:
        return {"error": str(e)}
